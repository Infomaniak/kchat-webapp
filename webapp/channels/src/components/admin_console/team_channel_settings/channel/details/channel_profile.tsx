// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import classNames from 'classnames';
import React from 'react';
import {FormattedMessage, defineMessage} from 'react-intl';

import type {Channel} from '@mattermost/types/channels';
import type {Team} from '@mattermost/types/teams';

import FormattedMarkdownMessage from 'components/formatted_markdown_message';
import SharedChannelIndicator from 'components/shared_channel_indicator';
import AdminPanel from 'components/widgets/admin_console/admin_panel';

import './channel_profile.scss';
interface ChannelProfileProps {
    channel: Partial<Channel>;
    team?: Team;
    onToggleArchive?: () => void;
    isArchived: boolean;
    isDisabled?: boolean;
}

export const ChannelProfile = (props: ChannelProfileProps): JSX.Element => {
    const {team, channel, isArchived, isDisabled} = props;

    const archiveBtn = isArchived ?
        defineMessage({id: 'admin.channel_settings.channel_details.unarchiveChannel', defaultMessage: 'Unarchive Channel'}) :
        defineMessage({id: 'admin.channel_settings.channel_details.archiveChannel', defaultMessage: 'Archive Channel'});

    let sharedBlock;
    if (channel.shared && channel.type) {
        sharedBlock = (
            <div className='channel-organizations'>
                <FormattedMarkdownMessage
                    id='admin.channel_settings.channel_detail.channelOrganizations'
                    defaultMessage='**Organizations**'
                />
                <br/>
                <SharedChannelIndicator
                    className='shared-channel-icon'
                    channelType={channel.type}
                />
                <FormattedMessage
                    id='admin.channel_settings.channel_detail.channelOrganizationsMessage'
                    defaultMessage='Shared with trusted organizations'
                />
            </div>
        );
    }

    return (
        <AdminPanel
            id='channel_profile'
            title={defineMessage({id: 'admin.channel_settings.channel_detail.profileTitle', defaultMessage: 'Channel Profile'})}
            subtitle={defineMessage({id: 'admin.channel_settings.channel_detail.profileDescription', defaultMessage: 'Summary of the channel, including the channel name.'})}
        >
            <div className='group-teams-and-channels AdminChannelDetails'>
                <div className='group-teams-and-channels--body channel-desc-col'>
                    <div className='channel-name'>
                        <FormattedMarkdownMessage
                            id='admin.channel_settings.channel_detail.channelName'
                            defaultMessage='**Name**'
                        />
                        <br/>
                        {channel.display_name}
                    </div>
                    <div className='channel-team'>
                        <FormattedMarkdownMessage
                            id='admin.channel_settings.channel_detail.channelTeam'
                            defaultMessage='**Team**'
                        />
                        <br/>
                        {team?.display_name}
                    </div>
                    {sharedBlock}
                    <div className='AdminChannelDetails_archiveContainer'>
                        <button
                            type='button'
                            className={
                                classNames(
                                    'btn',
                                    'btn-secondary',
                                    {'btn-danger': !isArchived},
                                    {disabled: isDisabled},
                                )
                            }
                            onClick={props.onToggleArchive}
                        >
                            {isArchived ?
                                <i className='icon icon-archive-arrow-up-outline'/> :
                                <i className='icon icon-archive-outline'/>}
                            <FormattedMessage {...archiveBtn}/>
                        </button>
                    </div>
                </div>
            </div>
        </AdminPanel>
    );
};
