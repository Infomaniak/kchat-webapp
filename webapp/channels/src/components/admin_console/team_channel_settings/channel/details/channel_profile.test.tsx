// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import type {Channel} from '@mattermost/types/channels';
import type {Team} from '@mattermost/types/teams';

import {ChannelProfile} from './channel_profile';

describe('admin_console/team_channel_settings/channel/ChannelProfile', () => {
    test('should match snapshot', () => {
        const testTeam: Partial<Team> = {display_name: 'test'};
        const testChannel: Partial<Channel> = {display_name: 'test'};
        const wrapper = shallow(
            <ChannelProfile
                isArchived={false}
                team={testTeam}
                channel={testChannel}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot for a shared channel', () => {
        const testTeam: Partial<Team> = {display_name: 'test'};
        const testChannel: Partial<Channel> = {
            display_name: 'test',
            type: 'O',
            shared: true,
        };
        const wrapper = shallow(
            <ChannelProfile
                isArchived={false}
                team={testTeam}
                channel={testChannel}
            />,
        );
        expect(wrapper).toMatchSnapshot();
    });
});
