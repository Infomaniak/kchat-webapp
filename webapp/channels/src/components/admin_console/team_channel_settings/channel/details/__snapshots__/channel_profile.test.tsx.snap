// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin_console/team_channel_settings/channel/ChannelProfile should match snapshot 1`] = `
<AdminPanel
  id="channel_profile"
  subtitle={
    Object {
      "defaultMessage": "Summary of the channel, including the channel name.",
      "id": "admin.channel_settings.channel_detail.profileDescription",
    }
  }
  title={
    Object {
      "defaultMessage": "Channel Profile",
      "id": "admin.channel_settings.channel_detail.profileTitle",
    }
  }
>
  <div
    className="group-teams-and-channels AdminChannelDetails"
  >
    <div
      className="group-teams-and-channels--body channel-desc-col"
    >
      <div
        className="channel-name"
      >
        <FormattedMarkdownMessage
          defaultMessage="**Name**"
          id="admin.channel_settings.channel_detail.channelName"
        />
        <br />
        test
      </div>
      <div
        className="channel-team"
      >
        <FormattedMarkdownMessage
          defaultMessage="**Team**"
          id="admin.channel_settings.channel_detail.channelTeam"
        />
        <br />
        test
      </div>
      <div
        className="AdminChannelDetails_archiveContainer"
      >
        <button
          className="btn btn-secondary btn-danger"
          type="button"
        >
          <i
            className="icon icon-archive-outline"
          />
          <MemoizedFormattedMessage
            defaultMessage="Archive Channel"
            id="admin.channel_settings.channel_details.archiveChannel"
          />
        </button>
      </div>
    </div>
  </div>
</AdminPanel>
`;

exports[`admin_console/team_channel_settings/channel/ChannelProfile should match snapshot for a shared channel 1`] = `
<AdminPanel
  id="channel_profile"
  subtitle={
    Object {
      "defaultMessage": "Summary of the channel, including the channel name.",
      "id": "admin.channel_settings.channel_detail.profileDescription",
    }
  }
  title={
    Object {
      "defaultMessage": "Channel Profile",
      "id": "admin.channel_settings.channel_detail.profileTitle",
    }
  }
>
  <div
    className="group-teams-and-channels AdminChannelDetails"
  >
    <div
      className="group-teams-and-channels--body channel-desc-col"
    >
      <div
        className="channel-name"
      >
        <FormattedMarkdownMessage
          defaultMessage="**Name**"
          id="admin.channel_settings.channel_detail.channelName"
        />
        <br />
        test
      </div>
      <div
        className="channel-team"
      >
        <FormattedMarkdownMessage
          defaultMessage="**Team**"
          id="admin.channel_settings.channel_detail.channelTeam"
        />
        <br />
        test
      </div>
      <div
        className="channel-organizations"
      >
        <FormattedMarkdownMessage
          defaultMessage="**Organizations**"
          id="admin.channel_settings.channel_detail.channelOrganizations"
        />
        <br />
        <SharedChannelIndicator
          channelType="O"
          className="shared-channel-icon"
        />
        <MemoizedFormattedMessage
          defaultMessage="Shared with trusted organizations"
          id="admin.channel_settings.channel_detail.channelOrganizationsMessage"
        />
      </div>
      <div
        className="AdminChannelDetails_archiveContainer"
      >
        <button
          className="btn btn-secondary btn-danger"
          type="button"
        >
          <i
            className="icon icon-archive-outline"
          />
          <MemoizedFormattedMessage
            defaultMessage="Archive Channel"
            id="admin.channel_settings.channel_details.archiveChannel"
          />
        </button>
      </div>
    </div>
  </div>
</AdminPanel>
`;
